src/action.ts

```diff
@@ -14,6 +14,8 @@ import { pipeline as streamPipeline } from "node:stream/promises"
 import AdmZip from "adm-zip"
 import { Readable } from "stream"
 import { randomUUID } from "crypto"
+import { mkdtemp } from "fs/promises"
+import { tmpdir } from "os"
```

Comment:
This is not needed either,  `mkdtempSync` already creates the directory in the OS `tmp` dir.